doc/dev/hmsl/hashicorp-vault.md

```diff
@@ -0,0 +1,82 @@
+# Hashicorp Vault
+
+The `ggshield check-secret-manager hashicorp-vault` command interacts with Hashicorp Vault instances.
+
+## Tests using cassettes
+
+Like other tests, a lot of the tests for ggshield.verticals.hmsl.secret_manager.hashicorp_vault.api_client
+are using cassettes recorded using [VCR.py](https://github.com/kevin1024/vcrpy) to replay network interactions.
+
+### Running Vault server locally with Docker
+
+To generate these cassettes, a local Hashicorp Vault instance in dev mode is used by running the official Vault docker image.
```

Comment:
This is a problem: ggshield has a policy of being able to recreate cassettes by just re-running tests (see https://github.com/GitGuardian/ggshield/blob/main/doc/dev/getting-started.md#L44). This is even enforced by the release script: it removes all cassettes and run the tests to verify they pass.

Need to think about the best approach for this 🤔.

Reply:
I see 3 possibilities here:

1. Create a fake server as a pytest fixture (looks like there aren't that many endpoints)
2. Mock HTTP calls instead of using cassettes (there is a `requests` mock module in `tests/unit/request_mock`)
3. Create a script to start a test server and populate it with test data; and another script to stop the test server. Make `scripts/release` call these scripts in its `run_test` function

Reply:
I started trying solution 3, I think it could work but it would be a bit time consuming to implement.

Solution 1/2 would work but wouldn't it be just cassettes but without the ease of recreating them?

Maybe we can skip the cassettes recreation for those tests?  Wdyt?

Reply:
Yeah I started working on the solution 3, I will continue investigating on how to create the test server properly .

Reply:
I implemented the dev server and modified the doc accordingly, so that all the tests that were using cassettes can be relaunched.